File: modify_backend.js

(() => {
    const fs = require("fs")
    const 原代码 = fs.readFileSync("src/calibre/db/backend.py", "utf8")
    let 替换后 = 原代码
    替换后 = 替换后.replace(替换后.match(
        '# Imports {{{')[0],
        '# Imports {{{\nimport re\n')
    let k = 2;
    while (--k) {
        替换后 = 替换后.replace(替换后.match(
            /author = ascii_filename.*/)[0],
            `author = re.sub(r"[\\/\\\\\\:\\*\\?\\"\\<\\>\\|]", "_", author.strip())[:l]`)
        替换后 = 替换后.replace(替换后.match(
            /title  = ascii_filename.*/)[0],
            `title  = re.sub(r"[\\/\\\\\\:\\*\\?\\"\\<\\>\\|]", "_", title.strip())[:l]`)
    }
    const outputPath = "src/calibre/db/backend_new.py"
    fs.writeFileSync(outputPath, 替换后)
    console.log('done: ', outputPath)
    process.exit(0)
})()

File: modify_backend.py
# encoding: utf8
def main():
    f = open('src/calibre/db/backend.py', 'r', encoding="utf-8")
    source = f.read()
    target = source
    target = target.replace(
        '# Imports {{{', '# Imports {{{\nimport re\n'
    )
    target = target.replace(
        'author = ascii_filename(author)[:l]',
        'author = re.sub(r"[\\/\\\\\\:\\*\\?\\"\\<\\>\\|]", "_", author.strip())[:l]'
    )
    if(source == target):
        return("WARNING: REPLACE AUTHOR FILENAME FAILS")

    source = target
    target = target.replace(
        'title  = ascii_filename(title.lstrip())[:l].rstrip()',
        'title  = re.sub(r"[\\/\\\\\\:\\*\\?\\"\\<\\>\\|]", "_", title.strip())[:l]'
    )
    if(source == target):
        return("WARNING: REPLACE TITLE FILENAME FAILS")
    f.close()

    f = open('src/calibre/db/backend_new.py', 'w', encoding="utf-8")
    f.write(target)
    return 'done'


if __name__ == "__main__":
    print(main())
